### PR TITLE
Feature/tet 150

### PR DESCRIPTION
**Pull request by vfarah-if**
### Checklist

* [ ] If this is a releasable change, has a news fragment been added?

  <details>
  <summary>Explanation</summary>
  
  A news fragment is required for any releasable change (i.e. code that runs in or affects production) so that a corresponding changelog entry is added when releasing.
  
  Check [changelog/README.md](https://github.com/uktrade/data-hub-api/blob/master/changelog/README.md) for instructions.
  
  </details>
  
* [ ] Has this branch been rebased on top of the current `develop` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [ ] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/develop/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

File: datahub/core/queue.py [hunk 1]
```diff
@@ -56,8 +70,108 @@ def clear(self):
...
+    def remove_scheduled_jobs(self, queue: str, job_or_ids):
+        if job_or_ids is not None:
+            registry = ScheduledJobRegistry(name=queue, connection=self._connection)
+            if isinstance(job_or_ids, Iterable):
+                for item in job_or_ids:
+                    registry.remove(item)
+                    logger.info(f'"Removing {item}')
+            else:
+                logger.info(f'"Removing {job_or_ids}')
+                registry.remove(job_or_ids)
```
Comment (vfarah-if) [inline datahub/core/queue.py:87]: Add tests for the single scenario

File: datahub/core/test/test_queue.py [hunk 2]
```diff
@@ -90,3 +83,98 @@ def test_cleans_up_redis_connection():
...
+    job_scheduler(
+        function=Spy.queue_handler,
+        function_args=('arg1', 'arg2'),
+        function_kwargs={'test': True},
+        queue_name='234',
+        is_burst=True,
+    )
+
+    retry_arg = datahub_queue_mock.call_args_list[-1][-1]['retry']
+    assert retry_arg is not None
```
Comment (vfarah-if) [inline datahub/core/test/test_queue.py:116]: Resolve getting retry reference less hacky

File: docker-compose.yml [hunk 3]
```diff
@@ -36,14 +36,16 @@ services:
...
-    command: watchmedo auto-restart -d . -R -p '*.py' -- python short-running-worker.py long-running-worker.py
+    command: python short-running-worker.py long-running-worker.py
 
   postgres:
     image: postgres:12
     restart: always
     environment:
       - POSTGRES_DB=datahub
       - POSTGRES_PASSWORD=datahub
+    ports:
```
Comment (vfarah-if) [inline docker-compose.yml:47]: Remove ports

File: datahub/search/test/test_tasks.py [hunk 4]
```diff
@@ -46,22 +45,12 @@ def test_sync_all_models(monkeypatch):
...
-def test_sync_object_task_retries_on_error(monkeypatch, opensearch):
-    """Test that the object task retries on error."""
-    sync_object_mock = Mock(side_effect=[Exception, None])
-    monkeypatch.setattr('datahub.search.sync_object.sync_object', sync_object_mock)
-
-    sync_object_task.apply_async(args=(SimpleModelSearchApp.name, str(uuid4())))
-
-    assert sync_object_mock.call_count == 2
-
-
```
Comment (vfarah-if) [inline datahub/search/test/test_tasks.py:64]: This is being tested from test queue and works different to Celery

**Review by marijnkampf — APPROVED** [review]

File: config/settings/test.py [hunk 5]
```diff
@@ -178,3 +178,5 @@
         MIDDLEWARE.remove('axes.middleware.AxesMiddleware')
     if 'axes.backends.AxesBackend' in AUTHENTICATION_BACKENDS:
         AUTHENTICATION_BACKENDS.remove('axes.backends.AxesBackend')
+
+IS_TEST = True
```
Comment (elcct) [inline config/settings/test.py:182]: Is this needed? It seems like you could just pass appropriate parameters during tests rather than modifying behaviour based on essentially a global variable.

File: datahub/core/queues/job_scheduler.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+            args=function_args,
+            kwargs=function_kwargs,
+            retry=Retry(
+                max=max_retries,
+                interval=retry_intervals,
+            ),
+        )
+        logger.info(f'Generated job id "{job.id}" with "{job.__dict__}"')
+        if settings.IS_TEST:
+            queue.work(queue_name)
```
Comment (elcct) [inline datahub/core/queues/job_scheduler.py:74]: You could add `immediate` function parameter defaulting to `False`. That would indicate the task will be executed upon enqueueing if that parameter was `True`.

Like:

```
if immediate:
  queue.work(queue_name)
```

File: datahub/core/queues/job_scheduler.py [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        return job
+
+
+def calculate_retry_intervals(
```
Comment (elcct) [inline datahub/core/queues/job_scheduler.py:78]: This function is quite hard to follow - using mixed types for parameters does not help.

Maybe it would be good to add a comment showing what is the expected output of the function based on different parameters or adding a couple of unit tests.
